subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/GradleUserHomeTransformerExecutionHistoryRepository.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.transform;
+
+import org.gradle.internal.execution.history.ExecutionHistoryStore;
+
+public class GradleUserHomeTransformerExecutionHistoryRepository extends DefaultTransformerExecutionHistoryRepository {
+    public GradleUserHomeTransformerExecutionHistoryRepository(GradleUserHomeWorkspaceProvider workspaceProvider, ExecutionHistoryStore executionHistoryStore) {
```

Comment:
Should we distinguish between `GradleUserHome...` and `Project...` variants here, or should we distinguish `Immutable...` vs `Mutable...`?

Reply:
Using `Immutable...` and `Mutable...` is maybe better, since it describes the domain concept better. The scope these things live in are a consequence of the type they have.

I'll change it.

Reply:
Changed.